src/wconfig/WorkerConfig.cc

```diff
@@ -78,6 +77,18 @@ WorkerConfig::WorkerConfig(const util::ConfigStore& configStore)
       _bufferMaxTotalGB(configStore.getInt("transmit.buffermaxtotalgb", 41)),
       _maxTransmits(configStore.getInt("transmit.maxtransmits", 40)),
       _maxPerQid(configStore.getInt("transmit.maxperqid", 3)) {
+    try {
+        _mySqlConfig = mysql::MySqlConfig(configStore.getRequired("mysql.username"),
+                                          configStore.get("mysql.password"),
+                                          configStore.getRequired("mysql.socket"));
+    } catch (util::KeyNotFoundError& e) {
```

Comment:
Is this supposed to handle the case of missing  "mysql.socket"? I'd again try to avoid relying on exceptions for this expected failure.